RELEASING.md

```diff
@@ -24,95 +24,211 @@ tools must be installed.
 </settings>
 ```
 
+### Sonatype account
+
+The release artifacts will be deployed into Sonatype OSS with the Maven deploy
+plugin.
+You must have an account with Sonatype, and it must be associated
+with the JanusGraph org.
+https://issues.sonatype.org/browse/OSSRH-28274
+
+### Configure Maven for server passwords
+
+The release artifacts will be uploaded into a staging directory on Sonatype
+OSS. If you do not configure Maven with your server passwords, the Maven
+deploy plugin will run into a 401 Unauthorized error.
+
+https://maven.apache.org/guides/mini/guide-encryption.html
+
+* Create a master password: `mvn --encrypt-master-password`
+
+* Add the master password to `$HOME/.m2/settings-security.xml` 
+
+```xml
+<settingsSecurity>
+  <master>{master-passsword}</master>
+</settingsSecurity>
+```
+
+* Once the master password has been added to `$HOME/.m2/settings-security.xml` , encrypt a server password: `mvn --encrypt-password`
+
+* Create `$HOME/.m2/settings.xml` using your Sonatype username and encrypted server password
+
+```xml
+<settings>
+  <servers>
+    <server>
+      <id>ossrh</id>
+      <username>...</username>
+      <password>...</password>
+    </server>
+  </servers>
+</settings>
+```
+
+### GPG passphrase
+
+You can pass it as a command line parameter to Maven with `-Dgpg.passphrase=$GPG_PASS`.
+You can also encrypt the gpg key with `mvn --encrypt-password` and add it to `$HOME/.m2/settings.xml` as shown below.
+You may still have to enter your GPG passphrase once.
+Otherwise you will have to type in your gpg passphrase many times when prompted during the build.  
+
+```xml
+<settings>
+   <servers>   
+    <server>
+      <id>ossrh</id>
+        <username></username>
+        <password></password>
+    </server>
+  </servers>
+  <profiles>
+    <profile>
+      <id>gpg</id>
+      <properties>
+        <gpg.executable>gpg</gpg.executable>
+        <gpg.passphrase>{encrypted gpg passphrase}</gpg.passphrase>
+      </properties>
+    </profile>
+  </profiles>
+  <activeProfiles>
+    <activeProfile>gpg</activeProfile>
+  </activeProfiles>
+</settings>
+```
+
+
+Release Checklist
+-----------------
+
+- [ ] Start up new janusgraph-dev thread on release with suggestions on what should be included in the release
+- [ ] Make sure all PRs and issues added since last release are associated to the milestone
+- [ ] Complete all items associated with milestone or move to a new milestone if necessary
+- [ ] Write up a synopsis of changes made in release for release page and vote thread
+- [ ] validate all changes have been merged upstream
+- [ ] create janusgraph-dev vote thread and get required votes
+- [ ] Tag release
```

Comment:
This says that first a release VOTE thread should be started and afterwards the release should be tagged, but doesn't the VOTE thread already contain a link to the tag on GitHub? The artifacts are also already linked, aren't they? So, should the vote thread be created after those two steps?

Reply:
Checklist has been updated and should be clearer now.

Reply:
I'll create corresponding sections to the checklist and make sure they're in order to improve the flow. After that it should be ready for another pass.